nodejs/week2/app.js

```diff
@@ -0,0 +1,97 @@
+const express = require('express');
+const app = express();
+const port = process.env.PORT || 3000;
+const documents = require('./documents.json');
+
+// Support parsing JSON requests
+app.use(express.json());
+
+app.get('/', (req, res) => {
+  res.send('This is a search engine');
+});
+
+app.get('/search', async (req, res) => {
+  try {
+    queryValue = req.query.q;
+
+    if (queryValue) {
+      const filteredArray = findByQuery(documents, queryValue);
+
+      res.json(filteredArray);
+    } else {
+      res.json(documents);
+    }
+  } catch (err) {
+    res.status(500).json({ error: 'Server error' });
+  }
+});
+
+app.post('/search', async (req, res) => {
+  try {
+    requestBody = req.body;
```

Comment:
You should probably capture `req.body.fields` and send it down to `findByPost ` to make it work with request body